core/src/miner/mem_pool_types.rs

```diff
@@ -33,21 +33,17 @@ pub enum TxOrigin {
     Local,
     /// External transaction received from network
     External,
-    /// Transaction from retracted blocks
-    RetractedBlock,
```

Comment:
I'm not sure about it.
What happened for the transactions in the rejected proposal block?

Reply:
Thank you, this change is not desirable. Retracted transactions by one-block retraction due to consensus failure (or by randomized-leader-election) should be considered.

Reply:
@sgkim126 and @HoOngEe What do you think about removing transactions from Memory pool only when the transactions are included in a finalized block?

Then we don't need to add transactions that was removed from the Memory pool to the Memory pool again.

Reply:
There are two methods removing transactions from the mem pool. One is `remove` and the other is `remove_old`. There is only one explicit call `remove` which removes invalid transactions(txs from malicious users or verification failed txs). Also there is one explicit call `remove_old` in `chain_new_blocks` which removes old (the norm of old here is the accounts' sequences in txs are old) transactions.
This `old` transactions are determined by chain's accounts' sequences in the latest block. The latest block fetches best block so the transactions in blocks rejected by consensus will not be removed from the mem pool.